entrypoint.sh

```diff
@@ -55,6 +55,11 @@ if [ -n "$VAULT_ADDR" ]; then
     export VAULT_TOKEN
   fi
 
+  if [ -z "$VAULT_TOKEN" ] && [ -n "$VAULT_AUTH_ROLE" ]; then
```

Comment:
What other types of vault auth methods are being considered with this change?

Reply:
Really just AWS (specifically Lambda). I copied these names/idea from the Vault Lambda extension. So you _could_ use other auth providers, but we're just going to be using AWS

Reply:
Gotcha! My initial concern with this was just that since the only auth methods we have mounted are token, aws, and kubernetes, this will fail for everything except aws which may be unclear. Would you be able to add that as context to either the code or the readme?

Reply:
I've been meaning to go through this README and improve it anyway. I'll add some notes about the different authentication mechanisms in place and add some details on this feature I'm adding